lib/charms/opensearch/v0/opensearch_config.py

```diff
@@ -153,6 +153,19 @@ def set_node(
             self.CONFIG_YML, "plugins.security.ssl.transport.enforce_hostname_verification", True
         )
 
+        # security plugin rest API access
+        self._opensearch.config.put(
+            self.CONFIG_YML,
+            "plugins.security.restapi.roles_enabled",
+            ["all_access", "security_rest_api_access"],
+        )
+        # to use the PUT and PATCH methods of the security rest API
+        self._opensearch.config.put(
+            self.CONFIG_YML,
+            "plugins.security.unsupported.restapi.allow_securityconfig_modification",
```

Comment:
As this function expands, it might be worth storing a `default_node_config.yml` file somewhere and updating it all at once, instead of rewriting this file multiple times

Reply:
It was decided by Mykola some time ago that we should NOT have templates (that we fill say with jinja or other), but what we could do in a refactoring ticket is to update all those keys at once, by providing a dictionary. So that we write once and that we make it clear all the keys that must be written

Reply:
ok, that seems like an acceptable compromise. Why are we avoiding templates?

Reply:
The reasoning was that it was overkill, and to avoid adding yet another component to manage (i.e jinja)

Reply:
ok, thanks for the explanation!